### ISSUE TITLE
[apex] ApexCRUDViolation incorrectly cleared when WITH SECURITY_ENFORCED is commented out

### ISSUE THREAD

**Issue by yarhamjohn** [pmd/pmd#4897]
**Affects PMD Version:** 
7.0.0-rc3

**Rule:**
ApexCRUDViolation - https://pmd.github.io/pmd/pmd_rules_apex_security.html#apexcrudviolation

**Description:**
PMD incorrectly clears the validate CRUD permissions error when `WITH SECURITY_ENFORCED` is added to the query by commented out.

**Code Sample demonstrating the issue:**

This query is not highlighted by PMD because the query includes `WITH SECURITY_ENFORCED` (correct):
```java
public List<Account> myMethod()
{
    List<Account> accounts = [
        SELECT Id, Name
        FROM Account
        WITH SECURITY_ENFORCED
    ];
    return accounts;
}
```
![image](https://github.com/pmd/pmd/assets/7828988/885de2b7-376c-4424-b879-e019c153c932)

-------------------

This query is not highlighted by PMD even though the `WITH SECURITY_ENFORCED` statement is commented out (incorrect):
```java
public List<Account> myMethod2()
{
    List<Account> accounts = [
        SELECT Id, Name
        FROM Account
        // WITH SECURITY_ENFORCED
    ];
    return accounts;
}
```
![image](https://github.com/pmd/pmd/assets/7828988/a15d5db8-c428-4b0d-88df-30cb5ea2a1e6)

**Expected outcome:**

PMD should report a violation for the second example as the `WITH SECURITY_ENFORCED` statement is commented out. This is a false-negative.

**Running PMD through:** VSCode using https://marketplace.visualstudio.com/items?itemName=chuckjonas.apex-pmd (v0.6.2)

**Comment by jsotuyod** [comment]
This needs to be rechecked against PMD 7.0.0. As the way we parsed Apex was completely replaced, dropping Jorje in favor of aapex-parser + SummitAST (see https://docs.pmd-code.org/latest/pmd_release_notes_pmd7.html#changed-apex-support-replaced-jorje-with-fully-open-source-front-end)

**Comment by jsotuyod** [comment]
I can confirm the final version of PMD 7.0.0 doesn't reproduce this issue.
